packages/manager/src/views/DashboardPage/components/Analytics/components/LineChart/LineChart.tsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import { useGetMonthlyDeploymentChart } from '@app/services/analytics';
+
+import { Card, CardTitle, CardBody, CardHeader } from '@patternfly/react-core';
+import { Chart, ChartAxis, ChartGroup, ChartLine, ChartThemeColor } from '@patternfly/react-charts';
+import { VictoryZoomContainer } from 'victory-zoom-container';
+
+interface IDeploymentData {
+  env: string;
+  count: number;
+  startDate: string;
+  endDate: string;
+}
+
+export const LineChart = () => {
+  const TotalMonthlyDeploymentData = useGetMonthlyDeploymentChart('', undefined, false).data;
+  const QAData = TotalMonthlyDeploymentData?.qa
```

Comment:
This is a very bad pattern. So this will get executed on each render all these sort and map.

Kindly write a react-query select function to group how you want and write a hook plugging in with it.

Reply:
Also with the majority of lines goes out and u can then move this component back to `packages/manager/src/views/DashboardPage/components/Analytics/Analytics.tsx` instead of seperate file